DeviceDetector.php

```diff
@@ -199,8 +199,8 @@ public function __construct(string $userAgent = '', ?ClientHints $clientHints =
         $this->addClientParser(new MobileApp());
         $this->addClientParser(new MediaPlayer());
         $this->addClientParser(new PIM());
-        $this->addClientParser(new Browser());
         $this->addClientParser(new Library());
+        $this->addClientParser(new Browser());
```

Comment:
can we restore the previous order position of the parsers?

Reply:
I guess the "problem" is, that PhantomJs isn't detected correctly then, as it always contains a valid browser in the user agent. But I agree that we should find another solution, that doesn't include changing the order. Not sure if it's worth to introduce a new detection class like `HeadlessBrowser` 🤔

Reply:
If we do that, then we get 3 failures:

```
1) DeviceDetector\Tests\DeviceDetectorTest::testParseClient with data set #561 (array('Android.LVLDM', array('library', 'Android License Verification Library', '')))
Fixture:
---
user_agent: Android.LVLDM
client:
  type: library
  name: Android License Verification Library
  version: ""

Result:
---
user_agent: Android.LVLDM
os:
  name: Android
  version: ""
  platform: ""
client:
  type: browser
  name: Android Browser
  version: ""
  engine: WebKit
  engine_version: ""
device:
  type: ""
  brand: ""
  model: ""
os_family: Android
browser_family: Android Browser

Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'type' => 'library'
-    'name' => 'Android License Verification Library'
+    'type' => 'browser'
+    'name' => 'Android Browser'
     'version' => ''
+    'engine' => 'WebKit'
+    'engine_version' => ''
 )

2) DeviceDetector\Tests\DeviceDetectorTest::testParseClient with data set #562 (array('Mozilla/5.0 (X11; Linux aarch...537.36', array('library', 'Cypress', '12.1.0')))
Fixture:
---
user_agent: Mozilla/5.0 (X11; Linux aarch64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/12.1.0 Chrome/106.0.5249.51 Electron/21.0.0 Safari/537.36
client:
  type: library
  name: Cypress
  version: 12.1.0

Result:
---
user_agent: Mozilla/5.0 (X11; Linux aarch64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/12.1.0 Chrome/106.0.5249.51 Electron/21.0.0 Safari/537.36
os:
  name: GNU/Linux
  version: ""
  platform: ARM
client:
  type: browser
  name: Chrome
  version: 106.0.5249.51
  engine: Blink
  engine_version: 106.0.5249.51
device:
  type: desktop
  brand: ""
  model: ""
os_family: GNU/Linux
browser_family: Chrome

Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'type' => 'library'
-    'name' => 'Cypress'
-    'version' => '12.1.0'
+    'type' => 'browser'
+    'name' => 'Chrome'
+    'version' => '106.0.5249.51'
+    'engine' => 'Blink'
+    'engine_version' => '106.0.5249.51'
 )

3) DeviceDetector\Tests\DeviceDetectorTest::testParseClient with data set #563 (array('Mozilla/5.0 (Unknown; Linux x.../538.1', array('library', 'PhantomJS', '2.1.1')))
Fixture:
---
user_agent: Mozilla/5.0 (Unknown; Linux x86_64) AppleWebKit/538.1 (KHTML, like Gecko) PhantomJS/2.1.1 Safari/538.1
client:
  type: library
  name: PhantomJS
  version: 2.1.1

Result:
---
user_agent: Mozilla/5.0 (Unknown; Linux x86_64) AppleWebKit/538.1 (KHTML, like Gecko) PhantomJS/2.1.1 Safari/538.1
os:
  name: GNU/Linux
  version: ""
  platform: x64
client:
  type: browser
  name: Safari
  version: ""
  engine: WebKit
  engine_version: "538.1"
device:
  type: desktop
  brand: ""
  model: ""
os_family: GNU/Linux
browser_family: Safari

Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    'type' => 'library'
-    'name' => 'PhantomJS'
-    'version' => '2.1.1'
+    'type' => 'browser'
+    'name' => 'Safari'
+    'version' => ''
+    'engine' => 'WebKit'
+    'engine_version' => '538.1'
 )
```